src/Validator.php

```diff
@@ -131,6 +131,10 @@ public function validate(
                 $validatedData = $dataSet->getAttributeValue($attribute);
                 $context->setParameter(ValidationContext::PARAMETER_VALUE_AS_ARRAY, null);
                 $context->setAttribute($attribute);
+
+                if ($dataSet instanceof LabelsProviderInterface) {
+                    $context->setAttributeLabel($dataSet->getValidationPropertyLabels()[$attribute] ?? null);
```

Comment:
Shouldn't we add:

```php
$context->setAttributeLabel(null);
```
in the other part of condition similar to:

https://github.com/yiisoft/validator/blob/163a813ac1d5b7ba61949c9eed4e09c80c954038/src/Validator.php#L128

?

Not sure about this one, but it's related to `Nested` and `Each` rules combination.